### DISCUSSION TITLE
Adding export: true on routes makes tests to fail.

### DISCUSSION THREAD

**Discussion by hbarrientosg** [ElMassimo/js_from_routes#25] [Q&A] [answered]
Basically I follow what you have on the documentation but it breaks my tests.

I add the export property on my routes.rb. The route generation goes fine.
```
      namespace :settings do
        resource :users, only: %i[show new create update], export: true
```
but on my tests 
```
      before { get :show, params: { id: id } }
      it { is_expected.to render_template('settings/users/show') }
```
This is the error:
```
     ActionController::UrlGenerationError:
       No route matches {:action=>"show", :controller=>"settings/users", :host=>"lvh.me:5000", :id=>3, :protocol=>"http"}
```
I did not create an issue because I don't know if there is something that I miss or It's a bug. I saw on the rails documentation that the route will have a key with :export and value of true and this will be mandatory. What do you think?

Modifying the test to this, it makes it green
```
      before { get :show, params: { id: outcomes_setting.id, export: true } }
```

**Comment by ElMassimo** [comment, marked as answer]
Hi Humberto! Thanks for reporting, this is indeed a bug. 

As you mentioned, using `:export` in the route makes it a required default, which doesn't affect how the route is processed but it does affect controller specs.

I've released [2.0.5](https://rubygems.org/gems/js_from_routes/versions/2.0.5) to prevent this behavior.
Please upgrade and let me know how it goes 😃 @hbarrientosg

**Comment by hbarrientosg** [comment]
Hi Máximo,

I updated the gem but I'm still having the same error. :(

**Reply by ElMassimo** [reply]
If it's not loading, perhaps it's due to the outdated installation instructions:

```ruby
group :development do
  gem 'js_from_routes'
end
```

and you'll need to load it in the test environment as well:

```ruby
group :development, :test do
  gem 'js_from_routes'
end
```

If that doesn't work, perhaps you can try [debugging `js_from_routes`](https://maximomussini.com/posts/debugging-ruby-libraries/#opening-gems-) and see if the patch is being executed in the test environment?

**Reply by hbarrientosg** [reply]
It works, You are right I have to enable it on test too.

**Reply by ElMassimo** [reply]
Excellent, thanks!
